### PR TITLE
Updated snippets to use new Messages v1 objects

### PR DESCRIPTION
**Pull request by dragonmantank**
## Description

Adds in the new Messages v1 objects to the Messages snippets

## Motivation and Context

Messages v1 was released, so this updates the code to reflect those changes

## Testing Details

Tested manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING.md** document.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dragonmantank** [comment]
@superchilled Can you check the WhatsApp stuff especially, as those used custom templates that don't really match the API. I'm not 100% sure if those are even supported now, or should we drop those back to using fully custom objects?

File: messages/messenger/send-text.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ const FB_SENDER_ID = process.env.FB_SENDER_ID;
 const BASE_URL = process.env.BASE_URL;
 
 const Vonage = require('@vonage/server-sdk');
+const MessengerImage = require('@vonage/server-sdk/lib/Messages/MessengerImage');
```
Comment (superchilled) [inline messages/messenger/send-text.js]:
```suggestion
const MessengerText = require('@vonage/server-sdk/lib/Messages/MessengerText');
```

File: messages/messenger/send-text.js [hunk 2]
```diff
@@ -24,21 +25,17 @@ const vonage = new Vonage(
...
-	{ type: 'messenger', id: FB_SENDER_ID },
-	{
-		content: {
-			type: 'text',
-			text:
-				'This is a Facebook Messenger text message sent using the Messages API',
-		},
-	},
+vonage.messages.send(
+	new MessengerImage(
```
Comment (superchilled) [inline messages/messenger/send-text.js]:
```suggestion
	new MessengerText(
```

File: messages/whatsapp/send-text.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ const WHATSAPP_NUMBER = process.env.WHATSAPP_NUMBER
 const BASE_URL = process.env.BASE_URL
 
 const Vonage = require('@vonage/server-sdk')
+const WhatsAppImage = require('@vonage/server-sdk/lib/Messages/WhatsAppImage');
```
Comment (superchilled) [inline messages/whatsapp/send-text.js]:
```suggestion
const WhatsAppText = require('@vonage/server-sdk/lib/Messages/WhatsAppText');
```

File: messages/whatsapp/send-text.js [hunk 4]
```diff
@@ -20,15 +21,12 @@ const vonage = new Vonage({
...
-  { "type": "whatsapp", "number": TO_NUMBER },
-  { "type": "whatsapp", "number": WHATSAPP_NUMBER },
-  {
-    "content": {
-      "type": "text",
-      "text": "This is a WhatsApp Message text message sent using the Messages API"
-    }
-  },
+vonage.messages.send(
+  new WhatsAppImage(
```
Comment (superchilled) [inline messages/whatsapp/send-text.js]:
```suggestion
  new WhatsAppText(
```

**Review by superchilled — CHANGES_REQUESTED** [review]
Looks good overall. Just a few small changes needed.

**Comment by superchilled** [comment]
> @superchilled Can you check the WhatsApp stuff especially, as those used custom templates that don't really match the API. I'm not 100% sure if those are even supported now, or should we drop those back to using fully custom objects?

As far as I can tell, the WhatsApp custom and template messages look ok. I've never really tested any of these examples in live code, but when we updated the curl examples for Messages v1 a few months back, I got them ok'd by the Messages PM. I'll make a note to follow up post-live and do some double-checking/ testing on these message types specifically (tbh, I think we could improve the documentation around WhatsApp template and custom messages generally, so I'll roll this in to a larger piece of work to audit/ improve the docs in that area -- we can always amend/ remove any snippets as necessary as part of that).

**Comment by dragonmantank** [comment]
Awesome, thanks. Those typos fixes are corrected, so once approved I'll get that merged then try and remember how to update the docs site :P

**Comment by superchilled** [comment]
> Awesome, thanks. Those typos fixes are corrected, so once approved I'll get that merged then try and remember how to update the docs site :P

Thanks Chris. FYI -- we just need to create the snippets, we don't need to add the snippets to the docs, (Nick/ Edmond on the ADP team are tasked with doing that).

**Review by superchilled — APPROVED** [review]
LGTM

**Pull request merged**